### ISSUE TITLE
Enabled plasmoid update and Apdatefier is trying to downgrade from 2.5 to 2.4

### ISSUE THREAD

**Issue by Mshmaitelly** [exequtic/apdatifier#13]
![image](https://github.com/exequtic/apdatifier/assets/13006192/15c16b14-cec4-4e0a-9958-91e1eb296688)

**Comment by exequtic** [comment]
For some reason, the API is currently returning outdated data. At least for my plasmoid, it still shows version 2.4, even though I uploaded it a few hours ago. Here's why he's trying to downgrade.

**Comment by DarkXero-dev** [comment]
Yeah I can add to this that it's also detecting updates for some plasmoids when there are none .. And hitting update will not update plasmoids .. Yet maybe in the future as it's still in beta ? 

![img](https://i.imgur.com/GzsgTq3.png)

https://store.kde.org/p/2137418

**Comment by exequtic** [comment]
> hitting update will not update plasmoids

Do you mean when pressing the "Upgrade system" button?

**Comment by DarkXero-dev** [comment]
@exequtic 

Yes sir click that button or from right-click context menu just refreshes and does not update plasmoids just system packages/flatpaks/aur...

**Comment by exequtic** [comment]
Yes, I have not yet implemented the update of all plasmoids at once. Because first, we need to make sure that everything works correctly with upgrading one by one.
The current problem with the downgrade confirms that.

**Comment by DarkXero-dev** [comment]
@exequtic 

Understood.. Will wait for a fix. Not a problem, And great work, your Plasmoid is part of my [XeroLinux Rice](https://github.com/xerolinux/xero-layan-git) ;)

**Comment by exequtic** [comment]
@DarkXero-dev
Cool, thanks!

**Comment by DarkXero-dev** [comment]
@exequtic I would update readme to include `jq` and `xmlstarlet` as necessary pkgs for plasmoid updater to work.. Since those are not included with any kde meta pkg..

**Comment by exequtic** [comment]
The API data has been updated, so version 2.4 should no longer appear in the list of available updates.

**Comment by DarkXero-dev** [comment]
will that also work for other Plasmoids having an update ?

**Comment by exequtic** [comment]
I think so. The list of updates includes those plasmoids whose version in metadata.json differs from the version indicated on the website. Read [here](https://www.reddit.com/r/kde/comments/1c1jqrt/apdatifier_can_now_show_plasmoids_updates_and), I explained how it works.

**Comment by DarkXero-dev** [comment]
Cool thanks I can update it in my rice now.. Great job...

**Comment by exequtic** [comment]
As I wrote on Reddit:

> on the first search, you may have plasmoids that actually have no updates 

Examples of such plasmoids:
power-management-inhibition **1.8 -> null**
alphablack-control **8 -> 0.1**
kicker-apps-only-menu **null -> 1.0.0**
...And there were a few others as well.

They actually have no updates because either their version on the website or in the metadata.json file is either incorrect or completely missing.

If you see, for example:
application-title-bar **0.4.3 -> 0.5.0**
then there is clearly an update, because the creator of the plasmoid hasn't forgotten to update the version in the metadata.json file and the version on the website.

> In my script, there's a simple version comparison, if they differ (no matter if the current version is higher or lower), then the plasmoid goes into the list.

**Comment by DarkXero-dev** [comment]
@exequtic  still for WUnderground
![image](https://github.com/exequtic/apdatifier/assets/81101439/a002adea-c147-448d-bb61-3c48b8a5a033)

**Comment by exequtic** [comment]
Well, it seems the data is still coming in outdated, at least for some plasmoids. I guess we'll have to wait some more.

**Comment by exequtic** [comment]
I've made several changes:
1) Widgets that do not have a version specified on the kde-store or in the metadata.json file (null) will be ignored.
2) Version comparison has been changed. Now, a widget will only appear in the list of available updates when the version on kde-store is greater than the current one. The current issue with downgrading will no longer occur.
3) During a "Full System Upgrade", widgets will also be upgraded.
